gettext: Problem with libxml2

I'm having trouble building gettext on some ancient SuSE11; see attached build log.  Has anyone come across this before?  `gettext/package.py` has not changed since last August.

```
libtool: link: /gpfsm/dnb53/rpfische/spack-tools/lib/spack/env/gcc/gcc -fPIC -o .libs/msgcomm msgcomm-msgcomm.o  ./.libs/libgettextsrc.so /gpfsm/dnb53/rpfische/tmp/spack-stage/spack-stage-WhGuEm/gettext-0.19.8.1/gettext-tools/gnulib-lib/.libs/libgettextlib.so -lacl /usr/lib64/libxml2.so -ldl -lz -lm -lncurses /gpfsm/dnb53/rpfische/tmp/spack-stage/spack-stage-WhGuEm/gettext-0.19.8.1/gettext-tools/intl/.libs/libintl.so -lc -fopenmp -Wl,-rpath -Wl,/gpfsm/dnb53/rpfische/spack-tools/opt/spack/linux-suse_linux11-x86_64/gcc-5.3.0/gettext-0.19.8.1-odf5uuy4jjcxpn3tcuogw7liukyqmusr/lib
./.libs/libgettextsrc.so: undefined reference to `xmlOutputBufferGetContent'
./.libs/libgettextsrc.so: undefined reference to `xmlXPathSetContextNode'
collect2: error: ld returned 1 exit status
```

[spack-build.out.txt](https://github.com/LLNL/spack/files/717680/spack-build.out.txt)

--- a/var/spack/repos/builtin/packages/gettext/package.py
+++ b/var/spack/repos/builtin/packages/gettext/package.py
@@ -86,8 +86,8 @@ class Gettext(AutotoolsPackage):
         if '+libxml2' in spec:
             config_args.append('CPPFLAGS=-I{0}/include'.format(
                 spec['libxml2'].prefix))
-            config_args.append('LDFLAGS=-L{0}/lib -Wl,-rpath,{0}/lib'.format(
-                spec['libxml2'].prefix))
+            config_args.append('LDFLAGS=-L{0} -Wl,-rpath,{0}'.format(
+                spec['libxml2'].libs))
         else:
             config_args.append('--with-included-libxml')
 

--- a/var/spack/repos/builtin/packages/gettext/package.py
+++ b/var/spack/repos/builtin/packages/gettext/package.py
@@ -87,7 +87,7 @@ class Gettext(AutotoolsPackage):
             config_args.append('CPPFLAGS=-I{0}/include'.format(
                 spec['libxml2'].prefix))
             config_args.append('LDFLAGS=-L{0} -Wl,-rpath,{0}'.format(
-                spec['libxml2'].libs))
+                spec['libxml2'].libs.directories[0]))
         else:
             config_args.append('--with-included-libxml')
 

--- a/var/spack/repos/builtin/packages/gettext/package.py
+++ b/var/spack/repos/builtin/packages/gettext/package.py
@@ -84,7 +84,9 @@ class Gettext(AutotoolsPackage):
             config_args.append('--disable-curses')
 
         if '+libxml2' in spec:
-            config_args.append('--with-libxml2-prefix={0}'.format(
+            config_args.append('CPPFLAGS=-I{0}/include'.format(
+                spec['libxml2'].prefix))
+            config_args.append('LDFLAGS=-L{0}/lib -Wl,-rpath,{0}/lib'.format(
                 spec['libxml2'].prefix))
         else:
             config_args.append('--with-included-libxml')

